README.md

```diff
@@ -2,22 +2,17 @@
 
 [![actions status](https://github.com/PyO3/pyo3/workflows/CI/badge.svg)](https://github.com/PyO3/pyo3/actions)
 [![codecov](https://codecov.io/gh/PyO3/pyo3/branch/main/graph/badge.svg)](https://codecov.io/gh/PyO3/pyo3)
-[![crates.io](http://meritbadge.herokuapp.com/pyo3)](https://crates.io/crates/pyo3)
+[![crates.io](https://meritbadge.herokuapp.com/pyo3)](https://crates.io/crates/pyo3)
 [![minimum rustc 1.41](https://img.shields.io/badge/rustc-1.41+-blue.svg)](https://rust-lang.github.io/rfcs/2495-min-rust-version.html)
 [![dev chat](https://img.shields.io/gitter/room/nwjs/nw.js.svg)](https://gitter.im/PyO3/Lobby)
+[![contributing notes](https://img.shields.io/badge/contribute-on%20github-Green)](https://github.com/PyO3/pyo3/blob/main/Contributing.md)
 
-[Rust](http://www.rust-lang.org/) bindings for [Python](https://www.python.org/). This includes running and interacting with Python code from a Rust binary, as well as writing native Python modules.
+[Rust](http://www.rust-lang.org/) bindings for [Python](https://www.python.org/), including tools for creating native Python extension modules. Running and interacting with Python code from a Rust binary is also supported.
 
 * User Guide: [stable](https://pyo3.rs) | [main](https://pyo3.rs/main)
 
 * API Documentation: [stable](https://docs.rs/pyo3/) |  [main](https://pyo3.rs/main/doc)
 
-* Contributing Notes: [github](https://github.com/PyO3/pyo3/blob/main/Contributing.md)
-
-* Architecture Guide: [github](https://github.com/PyO3/pyo3/blob/master/Architecture.md)
-
-A comparison with rust-cpython can be found [in the guide](https://pyo3.rs/main/rust_cpython.html).
```

Comment:
I wanted to make this section at the top of the README shorter, so I moved the contributing notes link into a badge, put the architecture guide link down in the new contributing section below, and removed the rust-cpython comparison note from here.

(Users who are interested in understanding the differences can still find the comparison in the guide.)